### PR TITLE
feat(sensing-messages): add sensing messages

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

This PR adds the sensing messages.

Related discussions/issues:
- https://github.com/orgs/autowarefoundation/discussions/216
- Example GNSS/INS messages:
  - Applanix: https://github.com/clearpathrobotics/applanix_driver/blob/hydro-devel/applanix_msgs/msg/NavigationSolution.msg
  - SBG Ellipse: https://github.com/SBG-Systems/sbg_ros_driver/tree/master/msg

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
I've finalized this PR for the GNSS/INS related messages. I've put the localization design information to the readme file too but we can separate it later.

Some points:
- I didn't want to use the [sensor_msgs/IMU](https://docs.ros.org/en/api/sensor_msgs/html/msg/Imu.html) message because it lacks the linear velocity and angular acceleration fields.
- I didn't want to use [sensor_msgs/NavSatFix](https://docs.ros.org/en/api/sensor_msgs/html/msg/NavSatFix.html) because it has fields not used by the Autoware:
  - [sensor_msgs/NavSatStatus](https://docs.ros.org/en/api/sensor_msgs/html/msg/NavSatStatus.html) status
  - `uint8` position_covariance_type
  - `float64[9]` position_covariance (we are only using the diagonal)

Please review this and share your opinions.

File: autoware_sensing_msgs/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+In Autoware, only these variance values are used, mostly in the RMSE form. The rest of the covariance matrix is not used, can be left `0.0`.
+
+#### Example for TwistWithCovariance
+
+[This message](https://docs.ros.org/en/api/geometry_msgs/html/msg/TwistWithCovariance.html) contains the linear and angular velocities and the covariance matrix.
+
+Let's call RMSE values for these calculations as `σ_x, σ_y, σ_z, σ_r, σ_p, σ_y`.
+
+The covariance matrix can be constructed as follows:
+| <span> | <span> | <span> | <span> | <span> | <span> |
```
Comment (kenji-miyake) [inline autoware_sensing_msgs/README.md:202]: [style] Adding an empty line here enables auto-formatting of the table.

```suggestion

| <span> | <span> | <span> | <span> | <span> | <span> |
```

**Review by kenji-miyake — COMMENTED** [review]
At a glance, generally looks good. :+1: Thank you for adding these msgs!
The members of TIER IV's localization team will review the details.

**Comment by angry-crab** [comment]
LGTM.
The only concern is that [sensor_msgs/NavSatStatus](https://docs.ros.org/en/api/sensor_msgs/html/msg/NavSatStatus.html) might be needed if we want to drop some non-fixed messages.

**Comment by xmfcx** [comment]
> LGTM. My only concern is that [sensor_msgs/NavSatStatus](https://docs.ros.org/en/api/sensor_msgs/html/msg/NavSatStatus.html) might be needed if we want to drop some non-fixed messages.

It is the job of the driver/bridge to attach these values to the readings. If a message is non-fixed, they should not publish the message or assign very high RMSE values, depending on the confidence level.

File: autoware_sensing_msgs/README.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+- The linear field contains the linear velocities in the x, y, z axes.
+- The angular field contains the angular velocities in the x, y, z axes.
+- The covariance matrix parameters are linear and angular velocities in order.
+
+For detailed info about the covariance matrix [RMSE? Variances? Covariance Matrix?](#rmse-variances-covariance-matrix).
+
+### Acceleration
+
+For this information, you can use the [AccelWithCovariance](https://docs.ros.org/en/api/geometry_msgs/html/msg/AccelWithCovariance.html) message.
```
Comment (meliketanrikulu) [inline autoware_sensing_msgs/README.md:63]:
```suggestion
For this information, you can use the [AccelWithCovarianceStamped](https://docs.ros.org/en/api/geometry_msgs/html/msg/AccelWithCovarianceStamped.html) message.
```

**Review by meliketanrikulu — CHANGES_REQUESTED** [review]

**Review by meliketanrikulu — APPROVED** [review]

**Review by meliketanrikulu — APPROVED** [review]

**Comment by xmfcx** [comment]
@angry-crab I've replaced the GnssInsPosition message with the NavSatFix for compatibility. Could you review it again?

Also if @YamatoAndo @mitsudome-r @yukkysaito could review it'd be nice, some other tasks are blocked by this.

**Review by yukkysaito — APPROVED** [review]
sorry for late :pray: 
LGTM

**Pull request merged**